arangod/Replication2/ReplicatedLog/Components/StateHandleManager.cpp

```diff
@@ -69,16 +74,32 @@ StateHandleManager::StateHandleManager(
 void StateHandleManager::becomeFollower(
     std::unique_ptr<IReplicatedLogFollowerMethods> ptr) {
   auto guard = guardedData.getLockedGuard();
+  // The stateHandle is initialized as non-null, it can be reset
+  // to null via resign. As becomeFollower is part of the
+  // single threaded setup process, no one could have called
+  // resign.
+  TRI_ASSERT(guard->stateHandle != nullptr)
+      << "We did hand out references to StateHandleManager before completing "
+         "the setup";
   guard->stateHandle->becomeFollower(std::move(ptr));
 }
 
 void StateHandleManager::acquireSnapshot(const ParticipantId& leader,
                                          std::uint64_t version) noexcept {
-  auto guard = guardedData.getLockedGuard();
-  // TODO is the correct log index required here?
-  guard->stateHandle->acquireSnapshot(leader, LogIndex{0}, version);
+  if (auto guard = guardedData.getLockedGuard(); guard->stateHandle) {
+    // TODO is the correct log index required here?
```

Comment:
This is fine for now, the `LogIndex` is not even used.

Reply:
I moved the comment only ;)